src/Signature.ts

```diff
@@ -7,19 +7,23 @@ import {
     signatureToString,
     stringToSignature,
 } from './eosjs-numeric';
+import { constructElliptic, PublicKey } from './eosjs-key-conversions';
 
 /** Represents/stores a Signature and provides easy conversion for use with `elliptic` lib */
 export class Signature {
-
-    constructor(private signature: Key) {}
+    constructor(private signature: Key, private ec: EC) {}
```

Comment:
Since this wasn't a required parameter before, is it possible to add a default?

Reply:
It should be set during the creation of the Signature and it depends on whether the Signature is a k1, r1, or wa type, which is derived from fromString or fromElliptic operations.

Reply:
Constructing the elliptic object is an expensive operation so if a default is set (secp256k1 is the most likely default) then it would cause two elliptic operations if the type is not k1.

Reply:
Got it 👍 Since this is already a major version change, its a good time to break the API, but I'm just generally hesitant to without a good reason. Sounds like a good enough reason to though.